pushDockerImage should throw an exception if pushing image fails

**Steps to reproduce:**
`./gradlew pushDockerImage --debug` without necessary credentials

**Actual:**
`BUILD SUCCESSFUL` without any image actually pushed

**Expected:**
`BUILD FAILED`
bonus points for why it failed

**Build output:**

```
...
2023-05-22T12:08:59.880+0200 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.hc.client5.http.wire] http-outgoing-8 >> "POST /images/ghcr.io%2Fxenit-eu%2F[redacted]:0.1.0-SNAPSHOT/push HTTP/1.1[\r][\n]"
2023-05-22T12:08:59.880+0200 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.hc.client5.http.wire] http-outgoing-8 >> "X-Registry-Auth: eyJ1c2VybmFtZSI6bnVsbCwicGFzc3dvcmQiOm51bGwsImVtYWlsIjpudWxsLCJzZXJ2ZXJhZGRyZXNzIjoiaHR0cHM6Ly9pbmRleC5kb2NrZXIuaW8vdjEvIiwiYXV0aCI6bnVsbCwicmVnaXN0cnl0b2tlbiI6bnVsbCwiaWRlbnRpdHl0b2tlbiI6bnVsbCwic3RhY2tPcmNoZXN0cmF0b3IiOm51bGx9[\r][\n]"

...
2023-05-22T12:09:00.724+0200 [DEBUG] [com.bmuschko.gradle.docker.shaded.org.apache.hc.client5.http.wire] http-outgoing-8 << "{"errorDetail":{"message":"unauthorized: unauthenticated: User cannot be authenticated with the token provided."},"error":"unauthorized: unauthenticated: User cannot be authenticated with the token provided."}[\r][\n]"
...
2023-05-22T12:09:00.725+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Execute start for :pushDockerImage'
2023-05-22T12:09:00.726+0200 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Completing Build operation 'Executing task ':pushDockerImage''
...
2023-05-22T12:09:00.732+0200 [LIFECYCLE] [org.gradle.internal.buildevents.BuildResultLogger] 
2023-05-22T12:09:00.732+0200 [LIFECYCLE] [org.gradle.internal.buildevents.BuildResultLogger] BUILD SUCCESSFUL in 3s
```

--- a/build.gradle
+++ b/build.gradle
@@ -129,7 +129,7 @@ dependencies {
         exclude group: 'org.slf4j'
     }
 
-    implementation 'com.bmuschko:gradle-docker-plugin:9.3.1'
+    implementation 'com.bmuschko:gradle-docker-plugin:9.3.2'
     implementation 'com.avast.gradle:gradle-docker-compose-plugin:0.17.7'
     testImplementation group: 'junit', name: 'junit', version: '4.13.2'
     testImplementation 'org.hamcrest:hamcrest-core:3.0'

